src/lib.rs

```diff
@@ -1274,6 +1267,11 @@ where
         let options = GetOptions::new().with_range(Some(range));
         self.get_opts(location, options).await?.bytes().await
     }
+
+    async fn head(&self, location: &Path) -> Result<ObjectMeta> {
+        let options = GetOptions::new().with_head(true);
```

Comment:
TIL that GetOpts already has `with_head` 🤯 

The more I see of these PRs to extract code into ObjectStoreExt the better I like it